### ISSUE TITLE
Godot 4.2.1+ AnimationTree advance expression in the editor is not parsing C# enum or const.

### ISSUE THREAD

**Issue by Akanaro** [godotengine/godot#87916]
### Tested versions

- Reproducible in 4.2.1 Mono and 4.2.2 RC1 Mono.

### System information

Godot v4.2.2.rc1.mono - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 660 Ti (NVIDIA; 30.0.14.7464) - Intel(R) Core(TM) i7-2600 CPU @ 3.40GHz (8 Threads)

### Issue description

Using an enum or const in the advance expression field on the AnimationTree node inside the editor does not work when using C# scripts. You should be able to simply add something like "state == IDLE".

### Steps to reproduce

Using an enum or const in the advance expression field on the AnimationTree node inside the editor does not work. For example in script you add something like:

`public enum States {IDLE, WALK, RUN, JUMP}`

or

```
const int IDLE = 0;
const int WALK = 1;
const int RUN = 2;
const int JUMP = 3;
```
and you should be able to simply use:

`state == States.IDLE`

or

`state == IDLE`

but it does not work. The only thing that works is directly using:

`state == 1`

for example, which is pointless. If you use the same setup in GDScript it works fine and if forgo the editor and use pure code it also works so the problem lies in the way the editor is parsing C# values.

### Minimal reproduction project (MRP)

N/A

**Comment by Calinou** [comment]
This is likely because the Expression class isn't aware of C# enums. Implementing that sounds pretty challenging.

**Comment by Akanaro** [comment]
Yeah, I can imagine it's not an easy task and I guess the use case is also not that common. Most folks using C# just code their own state machine from scratch instead of relying on expressions. Well, hopefully at some point the Expression class and the AnimationTree  expressions system gets some love.

**Comment by EllaLuna** [comment]
@Akanaro Thank you for this, I thought I was going crazy and this actually helped a lot.
Hopefully will be fixed at some point.
